### PR TITLE
test(testclient): migrate TC2-02 to managed clients

### PR DESCRIPTION
**Pull request by Katsiarynakavaleuskaya**
# Pull Request

## Goal

Migrate the bounded TC2-02 test cohort to the canonical managed TestClient lifecycle while preserving endpoint, auth, limiter, provider, database-isolation, active-suite, and explicitly ignored-suite contracts.

## Business reason

Reduce test false-greens and cross-test lifecycle leakage so each later TestClient cohort can be planned from a verified fresh-main boundary.

## Scope

- Replace direct or deprecated client ownership in 18 bounded test callers with the canonical shared fixture or context-managed `tests._client.open_test_client(...)`.
- Keep the three remaining local clients limited to isolated custom FastAPI mini-apps.
- Tighten affected auth, error, media-type, and deterministic contract assertions.
- Regenerate `.secrets.baseline` after the ordinary recovered-main integration.

## Out of scope

- No production runtime, OpenAPI, route, middleware, limiter, database-isolation, or pytest collection-policy changes.
- No changes to the main-recovery lanes.
- No TC2-03 through TC2-10 caller migrations in this PR.
- Per operator direction, this task stops after TC2-02 merge and exact-main sanity; it does not start the next PR lane.

## Files changed

- 18 test files under `tests/` and `tests/disabled_hypothesis/`.
- Generated `.secrets.baseline`.
- Exact branch diff: 19 paths; path-manifest SHA-256 `a17c2567edf01052a1836ce1b3d2f6c42e6f18147464ec06d2a5a84aad680621`.

## Key decisions

- Preserve the ordinary integration of recovered main `6a8aabc4b8a3f27b1a3eb363276d6498dc33ada4`; no rebase or history rewrite.
- Use the canonical shared client for the canonical application and managed local ownership only for custom mini-apps.
- Preserve the explicit `disabled_hypothesis` cohort as separately executed evidence.
- Keep legacy `_get_api_key_dynamic` alias coverage distinct from canonical `require_pro_tier` coverage.
- Keep Experiment Runner attribution non-material; no co-author trailer.

## Tests

Evidence bound to exact material head `177dd85a62c8bf9a39d004b23060e59f896d08be`:

- `python3 scripts/orchestration/check_preflight.py` — PASS.
- `python3 scripts/orchestration/check_agent_consistency.py` — PASS.
- `make validate-changed` — 224/224 PASS.
- `pre-commit run --all-files` and pre-push hooks — PASS without skipped hooks.
- Active cohort — 180/180 PASS.
- Explicit `disabled_hypothesis` cohort — 44/44 PASS.
- Lifecycle/provider/rate/no-direct/database/HTTPX/repo-policy bundle — 74/74 PASS.
- High-risk cohort — 67/67 forward and 67/67 reverse PASS.
- Security/TestClient guards — 116/116 PASS.
- Auth/tier guards — 138/138 PASS.
- Fresh strict Apple Container oracle — accepted; 20 + 180 + 44 PASS; `shared_tree_untouched=true`; `mutated_paths=[]`; immutable image digest `sha256:cb13886cf5c70a3d2659a83c4f8b0635c28c275f51f777a28f691b113a0ecfcb`.
- Ordered coordinator, QA, bug-hunter, and security exact-head roles — PROCEED; P0–P3 none.
- Exact-head canonical CI run `31353562058`: `test-pr (3.13)`, lint, coverage, diff-coverage, security, OpenAPI, scope and policy jobs completed successfully. The pre-closeout Merge readiness gate remains expected-red until this mapping-only descendant is published and review governance completes.

This evidence is not a merge-readiness claim. Mapping-head CI, thread resolution, the mandatory review wait cycle, and strict authenticated readiness remain pending.

## Security notes

No production or privileged runtime surface changes. Exact-head static audits found no raw canonical-app TestClient owner, new skip/xfail, suppression, network opt-in, database opt-in, or secret-baseline escape. The material seal records provider-neutral `review_claim=none` and `scan_claim=none`; provider absence is not review, scan, approval, PASS, or no-findings evidence.

## Risks / rollback

Primary risk is altered test setup/cleanup exposing a previously masked contract. Forward/reverse cohorts, restoration guards, exact-head CI, and the strict Apple oracle challenge that boundary. Rollback is a normal revert PR for this test-only slice; no production migration or data rollback is required.

## Discussion Thread Pass

- [x] Discussion-thread pass completed
- [x] Fixed in commit mapping completed

### Fixed in Commit Mapping

- [Canonical PR #2248 review mapping](https://github.com/Katsiarynakavaleuskaya/PulsePlate/blob/codex/testclient-tc2-02/docs/review/PR_2248_FIXED_MAPPING.md)

## Split justification

- TC2-02 is a precomputed caller cohort whose 18 test callers share one lifecycle/provider/limiter validation boundary.
- Moving the cohort together keeps the active/ignored totals, no-direct-client inventory, and reverse-order checks exact and auditable.
- TC2-03 through TC2-10 remain separate fresh-main slices; this task will only recommend the next slice after TC2-02 closure.

## Deferred / Follow-ups

- Do not start TC2-03 in this task.
- After exact-main TC2-02 closure, report the nearest recommended TC2-03 scope and the residual TC2-03…TC2-10 train plan.

## Next best step

Validate the uncommitted canonical mapping against the complete live bot inventory and the exact branch-ref body link, then publish the sole mapping-only closeout commit. After the mandatory review wait cycle and mapping-head CI, run strict authenticated merge readiness and merge normally.

## Summary by Sourcery

Перевести ограниченную тестовую когорту TC2-02 на управляемый жизненный цикл TestClient и ужесточить проверки эндпоинтов, аутентификации и error-contract, одновременно обновив baseline для secrets в соответствии с изменённой тестовой средой.

Bug Fixes:
- Синхронизировать тесты premium и VIP эндпоинтов с актуальными контрактами аутентификации и ответов об ошибках, включая стабильную семантику кодов 403/400/422.

Enhancements:
- Повысить детерминизм тестов и ясность поведения в областях недельного планирования, интеграции plate/targets, агрегации микронутриентов и предупреждений, зависящих от этапа жизни.
- Использовать управляемые жизненные циклы TestClient и общие фикстуры для всей когорты TC2-02, чтобы сократить ручное создание клиентов и мутацию среды.

Build:
- Перегенерировать `.secrets.baseline` после изменений в тестах и жизненном цикле клиента.

Tests:
- Перевести тесты TC2-02 на канонический поток `tests._client.open_test_client`, обеспечивая явное управление временем жизни клиента.
- Уточнить наборы disabled_hypothesis и крайних случаев для проверки точных полезных нагрузок ответов, результатов локализации и расчётов микронутриентов для premium nutrition эндпоинтов.
- Усилить покрытие эндпоинтов foods, BMI, bodyfat, weekly plan, plate, targets, VIP shoplist и insight при реалистичных конфигурациях API-ключей и feature-флагов.

Chores:
- Обновить исторические тестовые когорты в соответствии с восстановленным поведением основной ветки и текущими API-контрактами.

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Migrate the bounded TC2-02 test cohort to the managed TestClient lifecycle and tighten endpoint, auth, and error-contract assertions while refreshing the secrets baseline to reflect the updated test environment.

Bug Fixes:
- Align premium and VIP endpoint tests with current authentication and error response contracts, including stable 403/400/422 semantics.

Enhancements:
- Improve test determinism and clarity around weekly planning, plate/targets integration, micronutrient aggregation, and life-stage warning behavior.
- Use managed TestClient lifecycles and shared fixtures across the TC2-02 cohort to reduce ad-hoc client construction and environment mutation.

Build:
- Regenerate .secrets.baseline after test and client lifecycle changes.

Tests:
- Migrate TC2-02 tests to the canonical tests._client.open_test_client flow, ensuring explicit client lifetime management.
- Refine disabled_hypothesis and edge-case suites to assert exact response payloads, localization outputs, and micronutrient calculations for premium nutrition endpoints.
- Strengthen coverage of foods, BMI, bodyfat, weekly plan, plate, targets, VIP shoplist, and insight endpoints under realistic API key and feature-flag configurations.

Chores:
- Update historical test cohorts to match the recovered main branch behavior and current API contracts.

</details>

---
## Summary by cubic
Migrates the TC2‑02 test cohort to managed clients using the pytest `client` fixture and `tests._client.open_test_client`, tightens 422/JSON response contracts, removes flakiness with deterministic helpers, aligns the legacy weekly plan alias with canonical Pro headers, and adds the review governance closeout doc. No runtime, routes, or OpenAPI changes.

- **Refactors**
  - Replaced direct `fastapi.testclient.TestClient` usage and ad‑hoc `app` imports with the pytest `client` fixture and `tests._client.open_test_client` (including router‑only apps).
  - Bound function‑scoped clients via autouse fixtures; removed module‑level clients and casts.
  - Set API keys and feature flags per test with `monkeypatch.setenv`.
  - Regenerated `.secrets.baseline` with fewer findings.

- **Bug Fixes**
  - Standardized validation to 422 and asserted `application/json` content‑type across BMI, targets, plate, weekly plan, and VIP endpoints.
  - Added deterministic micronutrient and plate→targets helpers and a deterministic weekly‑plan seam; clarified targets‑only weekly‑plan error detail.
  - Enforced explicit tier/auth checks for premium/VIP paths, including canonical Pro headers for the legacy weekly plan alias; added missing i18n and header assertions.
  - Fail‑closed on foods router dependencies; tests patch `MeiliSearchBackend`/`ShadowSearchBackend` and assert guarded responses.

<sup>Written for commit b1aa73f9f5e8fdf5ed718322325537086463c488. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/Katsiarynakavaleuskaya/PulsePlate/pull/2248?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

- **Tests**
  - Improved validation of API status codes, authentication failures, response formats, and error details.
  - Expanded coverage for nutrition targets, micronutrient totals, weekly planning, BMI, body-fat, VIP access, and multilingual responses.
  - Increased reliability through isolated test sessions and deterministic scenarios.
  - Strengthened checks for API documentation availability, sanitized errors, and endpoint behavior across edge cases.
  - Updated security-scan baselines to reflect current findings.

### CODE REVIEW

File: tests/edges/test_premium_week_edges.py [hunk 1]
```diff
@@ -1,30 +1,36 @@
...
+@contextmanager
+def _make_pro_client() -> Iterator[TestClient]:
     from app.routers import pro
 
     app = FastAPI()
     app.dependency_overrides[require_pro_tier] = lambda: None
     app.include_router(pro.router)
-    return TestClient(cast(ASGIApp, app))
+    with open_test_client(app) as managed_client:
+        yield managed_client
```
[the remark this answers is not in the document]
Comment (Katsiarynakavaleuskaya) [reply tests/edges/test_premium_week_edges.py:31]: Disposition: FIXED
Commit: 177dd85a62c8bf9a39d004b23060e59f896d08be
Evidence: All four /api/v1/premium/plan/week-flexible calls now use pro_headers and execute the real require_pro_tier dependency; the exact nodes pass with asserted 400/400/200/200 outcomes, and TestRequireProTier passes 9/9. The sibling /api/v1/premium/{bmr,plate,targets} and /premium_targets tests intentionally retain their legacy _get_api_key_dynamic contract and are not PRO-tier bypasses.
Canonical mapping: https://github.com/Katsiarynakavaleuskaya/PulsePlate/blob/codex/testclient-tc2-02/docs/review/PR_2248_FIXED_MAPPING.md

File: tests/test_api_extras.py [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
 
 from app import app
+from tests._client import open_test_client
+
 
-client = TestClient(app)
+@pytest.fixture
+def api_extras_client() -> Iterator[TestClient]:
+    """Open one function-scoped managed client for the canonical app."""
+    with open_test_client(app) as managed_client:
```
[the remark this answers is not in the document]
Comment (Katsiarynakavaleuskaya) [reply tests/test_api_extras.py]: Disposition: FIXED
Commit: 99344f2171c466dce3e81fe2b7a9eee4675503eb
Evidence: Canonical-application tests now reuse the shared client fixture. Exact-head AST finds zero redundant canonical-app lifecycle owners and exactly three open_test_client calls, each owning an isolated custom FastAPI mini-app; provider/no-direct guards pass.
Canonical mapping: https://github.com/Katsiarynakavaleuskaya/PulsePlate/blob/codex/testclient-tc2-02/docs/review/PR_2248_FIXED_MAPPING.md
